### PR TITLE
Add new amount fields to the Order type

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it adds new fields to the order type as descibed here: https://github.com/saleor/saleor/pull/10350

# Impact

* [x] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/order/types.py [hunk 1]
```diff
@@ -1040,6 +1042,36 @@ class Order(ModelObjectType):
...
+        + ADDED_IN_38
+        + PREVIEW_FEATURE,
+        permissions=[OrderPermissions.MANAGE_ORDERS],
+    )
+    total_remaining_grant = PermissionsField(
+        Money,
+        required=True,
+        description=(
+            "The difference amount between granted refund and the "
+            "amounts that are pending and refunded.." + ADDED_IN_38 + PREVIEW_FEATURE
```
Comment (maarcingebala) [inline saleor/graphql/order/types.py:1071]:
```suggestion
            "amounts that are pending and refunded." + ADDED_IN_38 + PREVIEW_FEATURE
```

File: saleor/payment/models.py [hunk 2]
```diff
@@ -67,6 +67,15 @@ class TransactionItem(ModelWithMetadata):
         default=Decimal("0"),
     )
 
+    amount_pending_refund = MoneyField(
+        amount_field="pending_refund_value", currency_field="currency"
+    )
+    pending_refund_value = models.DecimalField(
```
Comment (maarcingebala) [inline saleor/payment/models.py:73]: Imo the naming here should follow the same pattern: pending_refund_amount, pending_refund_value.

**Review by fowczarek — APPROVED** [review]

File: saleor/payment/models.py (hunk 2 above)
Comment (korycins) [reply saleor/payment/models.py:73]: @maarcingebala I used the same pattern as we have for the rest of the fields:
```
amount_voided = MoneyField(amount_field="voided_value", currency_field="currency")
voided_value = models.DecimalField(
        max_digits=settings.DEFAULT_MAX_DIGITS,
        decimal_places=settings.DEFAULT_DECIMAL_PLACES,
        default=Decimal("0"),
    )
```

Comment (maarcingebala) [reply saleor/payment/models.py:73]: I see, but don't you think it's a bad pattern? :p

Comment (korycins) [reply saleor/payment/models.py:73]: Honestly, I don't have an opinion, but I would prefer to have the same naming convention for all fields. If you think that the field names should be changed then IMHO we should re-name all fields in separate PR.

Comment (korycins) [reply saleor/payment/models.py:73]: I will prepare a separate PR for renaming all fields in the same time. I will do that after we will merge the changes related to transaction

**Review by maarcingebala — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Review by jakubkuc — APPROVED** [review]

**Pull request merged**